roxie/ccd/ccdserver.cpp

```diff
@@ -11273,10 +11270,25 @@ class CRoxieServerXmlWriteActivity : public CRoxieServerDiskWriteActivity
         desc->queryProperties().setProp("@format","utf8n");
         desc->queryProperties().setProp("@rowTag",rowTag.get());
         desc->queryProperties().setProp("@kind", (kind==TAKjsonwrite) ? "json" : "xml");
-        if (headerLength)
-            desc->queryProperties().setPropInt("@headerLength", headerLength);
-        if (footerLength)
-            desc->queryProperties().setPropInt("@footerLength", footerLength);
+        if (kind==TAKjsonwrite)
+        {
+            if (headerLength)
+                desc->queryProperties().setPropInt("@headerLength", headerLength);
+            if (footerLength)
+                desc->queryProperties().setPropInt("@footerLength", footerLength);
+        }
+        else
+        {
+            IPartDescriptorIterator *parts = desc->getIterator();
+            if (parts->first())
+            {
+                do {
+                    IPartDescriptor &part = parts->query();
+                    part.queryProperties().setPropInt("@headerLength", headerLength);
+                    part.queryProperties().setPropInt("@footerLength", footerLength);
```

Comment:
Unfortunately it is not. 
In the special case when user wants to spray several "small" XML files (with different [size] header and/or footer – they are 100% valid XML files) into a multi slave Thor system (N:M). Our existing spray stores the header of the first XML file into the first part, then add content for several input  files (and drop out their header and footer) till reach the first part boundary, then gets the next XML file (e.g. 19th) and its header stores into the next part header and so on. And same thing is happening with footers, too.
(I faced this situation is couple of times in past 2 years). 
In this case each part can have a different (size) header/footer. That is why I stored lengths into the parts and use them for despray. 

I can imagine in the near future somebody wants to spray multiple small JSON file with different header/footer and the situation will be same.